### ISSUE TITLE
The new C# extension block syntax incorrectly triggers CA1515

### ISSUE THREAD

**Issue by rampaa** [dotnet/roslyn#81483]
**Version Used**: 
Visual Studio 2026 v18.0.0
.NET 10

**Steps to Reproduce**:

1. Unzip [FalsePositiveCA1515DueToExtensionBlock.zip](https://github.com/user-attachments/files/23832803/FalsePositiveCA1515DueToExtensionBlock.zip).
2. Compile the project and notice the warning: `CA1515: Consider making public types internal`.  
3. 
Comment out:  
```csharp
extension(System.Drawing.Rectangle rectangle)
{
    public Rect ToRect()
    {
        return new Rect(rectangle.X, rectangle.Y, rectangle.Width, rectangle.Height);
    }
}
```
and uncomment:
```csharp
public static Rect ToRect(this System.Drawing.Rectangle rectangle)
{
    return new Rect(rectangle.X, rectangle.Y, rectangle.Width, rectangle.Height);
}
```
4. Compile again and notice that it now compiles successfully without CA1515.

**Diagnostic Id**:

CA1515: Consider making public types internal

**Expected Behavior**:

The new extension block syntax should compile without triggering CA1515, just like the traditional extension method syntax.

**Actual Behavior**:

The new extension block syntax incorrectly triggers CA1515 while the traditional extension method syntax does not.

**Comment by rampaa** [comment]
To be clear, this issue has nothing to do with .editorconfig suppression not working. Marking this as the dupe of https://github.com/dotnet/roslyn-analyzers/issues/7734 makes more sense.

**Comment by jjonescz** [comment]
You're right, thanks!
